src/main/webapp/app/guided-tour/guided-tour.component.ts

```diff
@@ -173,68 +178,191 @@ export class GuidedTourComponent implements AfterViewInit, OnDestroy {
     /**
      * Scroll to and set highlighted element
      */
-    public scrollToAndSetElement(): void {
+    private scrollToAndSetElement(): void {
         this.selectedElementRect = this.updateStepLocation(this.getSelectedElement(), false);
         this.observeSelectedRectPosition();
 
         // Set timeout to allow things to render in order to scroll to the correct location
         setTimeout(() => {
-            if (this.isTourOnScreen()) {
+            if (this.isTourOnScreen(Direction.VERTICAL) && this.isTourOnScreen(Direction.HORIZONTAL)) {
                 return;
             }
-            const topPosition = this.getTopScrollingPosition();
-            try {
-                window.scrollTo({
-                    left: 0,
-                    top: topPosition,
-                    behavior: 'smooth',
-                });
-            } catch (err) {
-                if (err instanceof TypeError) {
-                    window.scroll(0, topPosition);
-                } else {
-                    throw err;
+            if (!this.isTourOnScreen(Direction.VERTICAL)) {
+                const topPosition = this.getTopScrollingPosition();
+                try {
+                    window.scrollTo({ left: 0, top: topPosition, behavior: 'smooth' });
+                } catch (err) {
+                    if (err instanceof TypeError) {
+                        window.scroll(0, topPosition);
+                    } else {
+                        throw err;
+                    }
                 }
             }
+            setTimeout(() => {
```

Comment:
I assume this timeout is needed to let the previously triggered scrolling finish? Is it not possible to move the flip orientation before the scrolling without timeout since the scrolling should not affect whether the tour is in the viewport horizontally?

Currently, the 300ms are noticeable and the user can see the popup jumping into the flipped orientation. Not super serious but if this could be avoided, even better.

Reply:
I changed the logic a little bit and now it's possible without the delay of 300ms :)